packages/flutter/test/material/menu_anchor_test.dart

```diff
@@ -2481,6 +2481,105 @@ void main() {
       );
     });
 
+    testWidgets('vertically constrained menus are positioned above the anchor with the provided offset', (WidgetTester tester) async {
+      await changeSurfaceSize(tester, const Size(800, 600));
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Builder(
+            builder: (BuildContext context) {
+              return Directionality(
+                textDirection: TextDirection.ltr,
+                child: Align(
+                  alignment: Alignment.bottomLeft,
+                  child: MenuAnchor(
+                    menuChildren: const <Widget> [
+                      MenuItemButton(child: Text('Button1'),
+                      ),
+                    ],
+                    builder: (BuildContext context, MenuController controller, Widget? child) {
+                      return FilledButton(
+                        onPressed: () {
+                          if (controller.isOpen) {
+                            controller.close();
+                          } else {
+                            controller.open();
+                          }
+                        },
+                        child: const Text('Tap me'),
+                      );
+                    },
+                  ),
+                ),
+              );
+            },
+          ),
+        ),
+      );
+
+      await tester.pump();
+      await tester.tap(find.text('Tap me'));
+      await tester.pump();
+
+      expect(find.byType(MenuItemButton), findsNWidgets(1));
+      // Test the default offset (0, 0) vertical position
+      expect(
+        collectSubmenuRects(),
+        equals(const <Rect>[
+          Rect.fromLTRB(0.0, 488.0, 122.0, 552.0),
+        ]),
+      );
+      // Closes the menu
+      await tester.tap(find.text('Tap me'));
+      await tester.pumpAndSettle();
+
+      await tester.pumpWidget(
```

Comment:
Nit: I would typically split this into 2 separate tests.  The main reason is better feedback when there is a failure in one of them.  But it's up to you if you think that's useful here.

Reply:
When does it make sense to have one test have multiple ```tester.pumpWidget``` widgets? In a previous PR, @TahaTesser told me to place both the default and explicit test cases in a single test since the separate test would only test the default behaviour.

Refactored nonetheless.

Reply:
> @TahaTesser told me to place both the default and explicit test cases in a single test since the separate test would only test the default behaviour.

That was the clip behavior test, which is a much simpler test.  Just checking the clip property.